Internal(Components): Components name overflow [ED-22032]

MuliElementor opened the pull request:
## PR Checklist
<!-- 
Please check if your PR fulfills the following requirements:
**Filling out the template is required.** Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
 -->
- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md


## PR Type
What kind of change does this PR introduce?
<!-- Please check the one that applies to this PR using "x" with no spaces eg: [x]. -->
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

<!--start_gitstream_placeholder-->
### ✨ PR Description
```
Purpose: Fix component name text overflow in Components tab by implementing ellipsis truncation with tooltip for long component names [ED-22032]
Main changes:
- Replaced ListItemText with EllipsisWithTooltip component to prevent name overflow and show full text on hover
- Restructured ListItemButton layout with flexbox properties (minWidth: 0, flexGrow: 1) to enable proper text truncation
- Refactored component wrapper from Fragment to Stack and reorganized styling for improved overflow handling
```

_Generated by LinearB AI and added by gitStream._
<sub>AI-generated content may contain inaccuracies. Please verify before using.
💡 **Tip:** You can customize your AI Description using **Guidelines** [Learn how](https://docs.gitstream.cm/automation-actions/#describe-changes)</sub>
<!--end_gitstream_placeholder-->


[ED-22032]: https://elementor.atlassian.net/browse/ED-22032?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

MuliElementor committed:
component validation

MuliElementor committed:
Enhance ComponentItem: Integrate EllipsisWithTooltip for component name display and refactor layout with Stack. Introduce Indicator styled component for active/error state visualization.

MuliElementor committed:
Refactor ComponentItem: Remove Indicator styled component and simplify component name display with EllipsisWithTooltip. Clean up unused imports and code.

MuliElementor committed:
Merge branch 'main' into ED-22032-components-name-overflow

MuliElementor committed:
revert

MuliElementor committed:
update test

elementorbot committed:
Merge branch 'main' into ED-22032-components-name-overflow

elementorbot committed:
Merge branch 'main' into ED-22032-components-name-overflow

elementorbot committed:
Merge branch 'main' into ED-22032-components-name-overflow

elementorbot committed:
Merge branch 'main' into ED-22032-components-name-overflow

MuliElementor committed:
update

IshayMaya commented on packages/packages/core/editor-components/src/components/__tests__/components-tab.test.tsx:
```diff
@@ -19,6 +19,7 @@ import { SearchProvider } from '../components-tab/search-provider';
 
 jest.mock( '@elementor/editor-documents', () => ( {
 	setDocumentModifiedStatus: jest.fn(),
+	isDocumentDirty: jest.fn( () => false ),
```
probably because you move the click handler up, the "More actions" click bubbles up

elementorbot committed:
Merge branch 'main' into ED-22032-components-name-overflow

MuliElementor committed:
Refactor ComponentItem for improved layout and interaction

- Removed unused mock for isDocumentDirty in tests.
- Enhanced the layout of the ComponentItem component by wrapping ListItemButton in a Box for better alignment and spacing.
- Updated styles to ensure consistent display and interaction behavior.

This change aims to improve the user experience and maintainability of the component.

elementorbot committed:
Merge branch 'main' into ED-22032-components-name-overflow
